sys/include/net/gnrc/pktbuf.h

```diff
@@ -94,6 +94,44 @@ void gnrc_pktbuf_init(void);
 gnrc_pktsnip_t *gnrc_pktbuf_add(gnrc_pktsnip_t *next, const void *data, size_t size,
                                 gnrc_nettype_t type);
 
+/**
+ * @brief   Adds a new gnrc_pktsnip_t and its packet to the packet buffer.
+ *          The new snip is appended at the end of the current packet snip list.
+ *
+ * @warning **Do not** change the fields of the gnrc_pktsnip_t created by this
+ *          function externally. This will most likely create memory leaks or
+ *          not allowed memory access.
+ *
+ * @pre size < CONFIG_GNRC_PKTBUF_SIZE
+ *
+ * @param[in] head      Pointer to gnrc_pktsnip_t in the packet.
+ *                      Can point to NULL, then it will contain the new head on
+ *                      success.
+ * @param[in] data      Data of the new gnrc_pktsnip_t. If @p data is NULL no data
+ *                      will be inserted into `result`.
+ * @param[in] size      Length of @p data. If this value is 0 the
+ *                      gnrc_pktsnip::data field of the newly created snip will
+ *                      be NULL.
+ * @param[in] type      Protocol type of the gnrc_pktsnip_t.
+ *
+ * @return  0 on success
+ * @return  -ENOMEM, if no space is left in the packet buffer.
+ */
+static inline int gnrc_pktbuf_add_tail(gnrc_pktsnip_t **head,
+                                       const void *data, size_t size,
+                                       gnrc_nettype_t type)
+{
+    gnrc_pktsnip_t *snip = gnrc_pktbuf_add(NULL, data, size, type);
+
+    if (snip == NULL) {
+        return -ENOMEM;
+    }
+
+    *head = gnrc_pkt_append(*head, snip);
+
+    return 0;
```

Comment:
Ok [after having some stomach aches after a conversation in the Matrix chat], I am fine with this change. It is just a convenience function (which might also be usable elsewhere). Does it make sense to emphasize that the actual list order is meant, not the packet order (which is reversed in the reception case).

Reply:
Actually with 7a5a5a3b7eecc18b5e0b4586a4850e1acab18b56 we don't need this function anymore. (It could still be handy though).

The question is if we want to modify the `gnrc_pktsnip_t` like that - ideally we could have a blocking future where we don't have to copy user data into GNRC in sdsdssdsdsternal buffers, but just keep the pointer to it while sending.

However until then copying the multiple user snips into one GNRC snip saves some allocation overhead.